### PR TITLE
Feat/new-user-stories

### PR DESCRIPTION
**Pull request by ese-varo**
## User stories for this feature
- #### As a registered user, I want to see a list of all the registered users and if I already have a conversation with them
  - See a list of all the users
  - Have a numeric indicator of how many conversations have with that user
    - You can display the conversations and show the name of the same that redirects you to the chat
  - If the registered user does not have any conversation with the user, you can click to create the conversation and redirects to it

- #### As a registered user, when it creates a conversation it has to add the description
  - At the moment when the user creates the conversation, he/she has to register emoji, title and description
    - Title, emoji, and description are required
    - Error handling
    - If the emoji is not found or represents an error, save an emoji by default
  - All the conversations have a profile
    - The profile consists of a title, emoji, and description
    - If the conversations are private, only the participants can see the profile
    - If the conversations are public, everyone can see the profile

- #### As an unregistered user, I can see a list of the last 5 conversations to know if I am interested in registering
  - In the login or a separated page, I can see a list of the last 5 most active public conversations (that have a greater number of messages and are not more than a month ago)
  - He/she can only see the profile of the conversation

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,10 +1,7 @@
 class UsersController < ApplicationController
-  def start_conversation
-    @users = User.all
-  end
-
   def new
     @user = User.new
+    # get '*path', to: redirect('/')
```
Comment (jvargas98) [inline app/controllers/users_controller.rb]: maybe you can remove this comment if you don't need it anymore

File: app/javascript/packs/conversation_show.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  headers: {
+    'Content-Type': 'application/json'
+  },
+}
+
+window.addEventListener("load", () => {
+  const queryString = window.location
+  console.log(queryString.href)
+  // setInterval(get_messages(queryString.href), 3000); 
+  // setInterval(() => { get_messages(queryString.href); }, 3000);
```
Comment (jvargas98) [inline app/javascript/packs/conversation_show.js:13]: The same here

File: app/javascript/packs/conversation_show.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+function render_messages(data) {
+  console.log(data)
+}
+
+// async function getalarms() {
+//   const url = BASE_URL + "alarms.json";
+//   const response = await fetch(url, options);
+//   const data = await response.json();
+//   evaluateAlarms(data);
+// }
```
Comment (jvargas98) [inline app/javascript/packs/conversation_show.js:31]: Same

File: app/models/message_observer.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ class MessageObserver < ActiveRecord::Observer
   def after_create(message)
     message.conversation.users.each do |user|
       next if user == message.user
-      ConversationMailer.new_message_email(user, message.conversation).deliver_now
+      # ConversationMailer.new_message_email(user, message.conversation).deliver_now
```
Comment (jvargas98) [inline app/models/message_observer.rb:5]: maybe you can remove this comment, but I think you need it.

File: app/views/conversations/profile.html.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      <div class="row">
+        <h1 class="col">
+          <%= presenter.emoji %> 
+        </h1>
+        <div class="col">
+          <% if current_user %>
+            <%= link_to 'Back', @conversation, class: 'btn btn-secondary float-right' %>
+          <% else %>
+            <%= link_to 'Back', root_path, class: 'btn btn-secondary float-right' %>
+          <% end %>
```
Comment (jvargas98) [inline app/views/conversations/profile.html.erb]: I'm not sure but maybe a ternary operator will work, I think this because only change the path

**Review by jvargas98 — COMMENTED** [review]
Great job, I liked how you add the services and object queries. I left you some comments and I also wish you could implement partial views to clean up your views a little more.

File: app/views/conversations/profile.html.erb (hunk 5 above)
Comment (GregVaz) [reply app/views/conversations/profile.html.erb]: It's correct, you can change the path with a ternary operator

File: app/views/layouts/_alerts.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<% flash.each do |name, msg| %>
+  <% if name == 'success' %>
+    <%= content_tag :div, msg, class: "alert alert-primary" %>
+  <% else %>
+      <%= content_tag :div, msg, class: "alert alert-warning" %>
+  <% end %>
+<% end %>
```
Comment (GregVaz) [inline app/views/layouts/_alerts.html.erb]: Just an idea
If you want to reduce this code, you could put the name of the _type of alert_ as the suffix of alert, you could establish your own css class and that's all
`<%= content_tag :div, msg, class: "alert alert-#{name}"`

**Comment by LoginALee** [comment]
Good work, i liked how you implemented error handling :+1:

File: app/views/layouts/_alerts.html.erb (hunk 6 above)
Comment (ese-varo) [reply app/views/layouts/_alerts.html.erb]: I actually tried that way, but it seems that rails doesn't let you use create custom alerts as `danger, primary, success, info` things like that which are the classes that bootstrap provides you to add styles to alerts. That's why I'm using the else option as a fallback for other than `success` flash messages. 
In order to implement the approach that you mention I should create custom classes that could match with the rails flash options, may be I could give it a try :sweat_smile:

File: app/views/conversations/profile.html.erb (hunk 5 above)
Comment (ese-varo) [reply app/views/conversations/profile.html.erb]: I'll try that, thanks!

**Review by LoginALee — APPROVED** [review]

**Review by jvargas98 — APPROVED** [review]
LGTM 🚀

**Review by GregVaz — APPROVED** [review]

**Pull request merged**